bootstrap/src/main/java/io/lumigo/javaagent/LumigoAgent.java

```diff
@@ -52,6 +60,20 @@ private static boolean isDebugMode() {
     return Boolean.parseBoolean(value);
   }
 
+  private static String parseVersion() {
+    String version = LumigoAgent.class.getPackage().getImplementationVersion();
+    if (version != null) {
+      try {
+        Matcher pt = Pattern.compile("^lumigo\\-(.*)?\\-otel.*$").matcher(version);
```

Comment:
The pattern should be a static final variable, we need it virtually on every agent load and we can spare the thread-unsafe nullity check.